app/services/association_authorizer.rb

```diff
@@ -7,7 +7,7 @@ def self.authorized_associations(associations, klass_name = nil, should_raise_ex
       end
       klass_name ||= associations_klass
       permission = permission_name(klass_name, action, should_raise_exception)
-      associations.authorized(permission, associations_klass) if permission
+      associations.authorized(permission, klass_name) if permission
```

Comment:
This is what was causing the issue where non admin users wouldn't see their locs/orgs. @tbrisker notice `klass_name` is passed from the patch #3779 but it wasn't being used (only for creating the permission name)